### ISSUE TITLE
support merging junit reports where the same testsuite exists in multiple files

### ISSUE THREAD

**Issue by superflyxxi** [bhovhannes/junit-report-merger#226]
CircleCI supports running unit tests in parallel. CircleCI can split the work evenly based on testcase or classname time taken. By having it split by classnames, each junit.xml file will have unique classes.

However, ideally we'd want to split by testcase. Some test cases are longer than others. And CircleCI may choose to split a testsuite into two threads. Because of this, one thread will have junit.xml that contains half of the testcases, while another thread will generate a separate junit.xml with the remaining.

When the two get merged, there are two testsuite elements with the same name, but contain diffrerent testcases. Code analysis tools like SonarCloud do not treat this as a valid junit.xml.

Ideally, we should match testsuites by name and not blindly add them in. I can share some examples if need be.

**Comment by bhovhannes** [comment]
Hello

I think I understand what is the issue, but I have never used CircleCI and
I am not sure that my understanding is correct.
Having a failing example will help. Please share 2 small inputs and
expected output.

At a first glance this seems to be something feasible to fix.

**Comment by superflyxxi** [comment]
Thanks for the quick reply. I'll attach somewhat verbose sample junit xml files. The junit-0.xml and the junit-1.xml have the same test suites (`SimpleScaleTest` and `SimpleScaleTest::testScaling`) but two different test cases. And the final junit.xml has the two TestSuites repeated. I've only shared junit-0.xml and junit-1.xml which contain the duplicate testsuite.

[junit-xmls.zip](https://github.com/bhovhannes/junit-report-merger/files/10951174/junit-xmls.zip)

`npx jrm ./junit.xml ./junit-0.xml ./junit-1.xml ./junit-2.xml ./junit-3.xml ./junit-4.xml`

Note: I also noticed that test suites are repeated over and over even though it only appears in two files. I'll log separate issue for this.
